File: 通讯协议/ProtoGen/output/handlers/csharp_all/ActivityFavorHandler.cs
/*-----------------------------------------------
 *本文件由代码生成器自动生成，
 *千万不要修改本文件的任何代码，
 *修改的的任何代码都会被覆盖掉！
 --------------------------------------------------*/

using System.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Pomelo.DotNetClient;

namespace pomelo.area
{
	public class ActivityFavorHandler
	{
		private PomeloClient _socket;

		public DailyRechargeGetInfoResponse lastDailyRechargeGetInfoResponse { get { return _socket.GetLastResponse<DailyRechargeGetInfoResponse>(); } }
		public DailyRechargeGetAwardResponse lastDailyRechargeGetAwardResponse { get { return _socket.GetLastResponse<DailyRechargeGetAwardResponse>(); } }
		public SuperPackageGetInfoResponse lastSuperPackageGetInfoResponse { get { return _socket.GetLastResponse<SuperPackageGetInfoResponse>(); } }
		public SuperPackageBuyResponse lastSuperPackageBuyResponse { get { return _socket.GetLastResponse<SuperPackageBuyResponse>(); } }
		public SevenDayPackageGetInfoResponse lastSevenDayPackageGetInfoResponse { get { return _socket.GetLastResponse<SevenDayPackageGetInfoResponse>(); } }
		public SevenDayPackageAwardResponse lastSevenDayPackageAwardResponse { get { return _socket.GetLastResponse<SevenDayPackageAwardResponse>(); } }
		public DailyDrawInfoResponse lastDailyDrawInfoResponse { get { return _socket.GetLastResponse<DailyDrawInfoResponse>(); } }
		public DailyDrawResponse lastDailyDrawResponse { get { return _socket.GetLastResponse<DailyDrawResponse>(); } }
		public RecoveredInfoResponse lastRecoveredInfoResponse { get { return _socket.GetLastResponse<RecoveredInfoResponse>(); } }
		public RecoveredResponse lastRecoveredResponse { get { return _socket.GetLastResponse<RecoveredResponse>(); } }
		static ActivityFavorHandler()
		{
			EventTypes.RegistPushType("area.activityFavorPush.superPackageBuyPush", typeof(SuperPackageBuyPush));
			EventTypes.RegistRequestType("area.activityFavorHandler.dailyRechargeGetInfoRequest", typeof(DailyRechargeGetInfoRequest), typeof(DailyRechargeGetInfoResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.dailyRechargeGetAwardRequest", typeof(DailyRechargeGetAwardRequest), typeof(DailyRechargeGetAwardResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.superPackageGetInfoRequest", typeof(SuperPackageGetInfoRequest), typeof(SuperPackageGetInfoResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.superPackageBuyRequest", typeof(SuperPackageBuyRequest), typeof(SuperPackageBuyResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.sevenDayPackageGetInfoRequest", typeof(SevenDayPackageGetInfoRequest), typeof(SevenDayPackageGetInfoResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.sevenDayPackageAwardRequest", typeof(SevenDayPackageAwardRequest), typeof(SevenDayPackageAwardResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.dailyDrawInfoRequest", typeof(DailyDrawInfoRequest), typeof(DailyDrawInfoResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.dailyDrawRequest", typeof(DailyDrawRequest), typeof(DailyDrawResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.recoveredInfoRequest", typeof(RecoveredInfoRequest), typeof(RecoveredInfoResponse));
			EventTypes.RegistRequestType("area.activityFavorHandler.recoveredRequest", typeof(RecoveredRequest), typeof(RecoveredResponse));
		}
		public ActivityFavorHandler(PomeloClient socket)
		{
			this._socket = socket;

		}
		public void dailyRechargeGetInfoRequest(Action<PomeloException,DailyRechargeGetInfoResponse> cb,object option = null)
		{
			var request = new DailyRechargeGetInfoRequest();
			_socket.request<DailyRechargeGetInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as DailyRechargeGetInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void dailyRechargeGetAwardRequest(int awardId,Action<PomeloException,DailyRechargeGetAwardResponse> cb,object option = null)
		{
			var request = new DailyRechargeGetAwardRequest();
			request.awardId= awardId;
			_socket.request<DailyRechargeGetAwardResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as DailyRechargeGetAwardResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void superPackageGetInfoRequest(Action<PomeloException,SuperPackageGetInfoResponse> cb,object option = null)
		{
			var request = new SuperPackageGetInfoRequest();
			_socket.request<SuperPackageGetInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as SuperPackageGetInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void superPackageBuyRequest(int packageId,int channelId,string c2s_imei,int c2s_os,Action<PomeloException,SuperPackageBuyResponse> cb,object option = null)
		{
			var request = new SuperPackageBuyRequest();
			request.packageId= packageId;
			request.channelId= channelId;
			request.c2s_imei= c2s_imei;
			request.c2s_os= c2s_os;
			_socket.request<SuperPackageBuyResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as SuperPackageBuyResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void sevenDayPackageGetInfoRequest(Action<PomeloException,SevenDayPackageGetInfoResponse> cb,object option = null)
		{
			var request = new SevenDayPackageGetInfoRequest();
			_socket.request<SevenDayPackageGetInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as SevenDayPackageGetInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void sevenDayPackageAwardRequest(int packageId,Action<PomeloException,SevenDayPackageAwardResponse> cb,object option = null)
		{
			var request = new SevenDayPackageAwardRequest();
			request.packageId= packageId;
			_socket.request<SevenDayPackageAwardResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as SevenDayPackageAwardResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void dailyDrawInfoRequest(Action<PomeloException,DailyDrawInfoResponse> cb,object option = null)
		{
			var request = new DailyDrawInfoRequest();
			_socket.request<DailyDrawInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as DailyDrawInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void dailyDrawRequest(int timeType,int costType,Action<PomeloException,DailyDrawResponse> cb,object option = null)
		{
			var request = new DailyDrawRequest();
			request.timeType= timeType;
			request.costType= costType;
			_socket.request<DailyDrawResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as DailyDrawResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void recoveredInfoRequest(Action<PomeloException,RecoveredInfoResponse> cb,object option = null)
		{
			var request = new RecoveredInfoRequest();
			_socket.request<RecoveredInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as RecoveredInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void recoveredRequest(int id,int type,Action<PomeloException,RecoveredResponse> cb,object option = null)
		{
			var request = new RecoveredRequest();
			request.id= id;
			request.type= type;
			_socket.request<RecoveredResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as RecoveredResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void onSuperPackageBuyPush(Action<SuperPackageBuyPush> cb)
		{
			_socket.listen_once<SuperPackageBuyPush>(cb);
		}

	}


}

File: 通讯协议/ProtoGen/output/handlers/lua/ActivityFavorHandler.lua
-------------------------------------------------
-- 本文件由代码生成器自动生成，
-- 千万不要修改本文件的任何代码，
-- 修改的的任何代码都会被覆盖掉！
----------------------------------------------------

local Socket = require "Xmds.Pomelo.LuaGameSocket"
require "base64"
require "activityFavorHandler_pb"

--------------------------------Pomelo-------------------------------
Pomelo = Pomelo or {}

--------------------------------ActivityFavorHandler-------------------------------
Pomelo.ActivityFavorHandler = {}

local function dailyRechargeGetInfoRequestEncoder(msg)
	local input = activityFavorHandler_pb.DailyRechargeGetInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function dailyRechargeGetInfoRequestDecoder(stream)
	local res = activityFavorHandler_pb.DailyRechargeGetInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.dailyRechargeGetInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.activityFavorHandler.dailyRechargeGetInfoRequest", option)
	Socket.Request("area.activityFavorHandler.dailyRechargeGetInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastDailyRechargeGetInfoResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.dailyRechargeGetInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.dailyRechargeGetInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.dailyRechargeGetInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, dailyRechargeGetInfoRequestEncoder, dailyRechargeGetInfoRequestDecoder)
end


local function dailyRechargeGetAwardRequestEncoder(msg)
	local input = activityFavorHandler_pb.DailyRechargeGetAwardRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function dailyRechargeGetAwardRequestDecoder(stream)
	local res = activityFavorHandler_pb.DailyRechargeGetAwardResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.dailyRechargeGetAwardRequest(awardId,cb,option)
	local msg = {}
	msg.awardId = awardId
	Socket.OnRequestStart("area.activityFavorHandler.dailyRechargeGetAwardRequest", option)
	Socket.Request("area.activityFavorHandler.dailyRechargeGetAwardRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastDailyRechargeGetAwardResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.dailyRechargeGetAwardRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.dailyRechargeGetAwardRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.dailyRechargeGetAwardRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, dailyRechargeGetAwardRequestEncoder, dailyRechargeGetAwardRequestDecoder)
end


local function superPackageGetInfoRequestEncoder(msg)
	local input = activityFavorHandler_pb.SuperPackageGetInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function superPackageGetInfoRequestDecoder(stream)
	local res = activityFavorHandler_pb.SuperPackageGetInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.superPackageGetInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.activityFavorHandler.superPackageGetInfoRequest", option)
	Socket.Request("area.activityFavorHandler.superPackageGetInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastSuperPackageGetInfoResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.superPackageGetInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.superPackageGetInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.superPackageGetInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, superPackageGetInfoRequestEncoder, superPackageGetInfoRequestDecoder)
end


local function superPackageBuyRequestEncoder(msg)
	local input = activityFavorHandler_pb.SuperPackageBuyRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function superPackageBuyRequestDecoder(stream)
	local res = activityFavorHandler_pb.SuperPackageBuyResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.superPackageBuyRequest(packageId,channelId,c2s_imei,c2s_os,cb,option)
	local msg = {}
	msg.packageId = packageId
	msg.channelId = channelId
	msg.c2s_imei = c2s_imei
	msg.c2s_os = c2s_os
	Socket.OnRequestStart("area.activityFavorHandler.superPackageBuyRequest", option)
	Socket.Request("area.activityFavorHandler.superPackageBuyRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastSuperPackageBuyResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.superPackageBuyRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.superPackageBuyRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.superPackageBuyRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, superPackageBuyRequestEncoder, superPackageBuyRequestDecoder)
end


local function sevenDayPackageGetInfoRequestEncoder(msg)
	local input = activityFavorHandler_pb.SevenDayPackageGetInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function sevenDayPackageGetInfoRequestDecoder(stream)
	local res = activityFavorHandler_pb.SevenDayPackageGetInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.sevenDayPackageGetInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.activityFavorHandler.sevenDayPackageGetInfoRequest", option)
	Socket.Request("area.activityFavorHandler.sevenDayPackageGetInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastSevenDayPackageGetInfoResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.sevenDayPackageGetInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.sevenDayPackageGetInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.sevenDayPackageGetInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, sevenDayPackageGetInfoRequestEncoder, sevenDayPackageGetInfoRequestDecoder)
end


local function sevenDayPackageAwardRequestEncoder(msg)
	local input = activityFavorHandler_pb.SevenDayPackageAwardRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function sevenDayPackageAwardRequestDecoder(stream)
	local res = activityFavorHandler_pb.SevenDayPackageAwardResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.sevenDayPackageAwardRequest(packageId,cb,option)
	local msg = {}
	msg.packageId = packageId
	Socket.OnRequestStart("area.activityFavorHandler.sevenDayPackageAwardRequest", option)
	Socket.Request("area.activityFavorHandler.sevenDayPackageAwardRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastSevenDayPackageAwardResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.sevenDayPackageAwardRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.sevenDayPackageAwardRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.sevenDayPackageAwardRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, sevenDayPackageAwardRequestEncoder, sevenDayPackageAwardRequestDecoder)
end


local function dailyDrawInfoRequestEncoder(msg)
	local input = activityFavorHandler_pb.DailyDrawInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function dailyDrawInfoRequestDecoder(stream)
	local res = activityFavorHandler_pb.DailyDrawInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.dailyDrawInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.activityFavorHandler.dailyDrawInfoRequest", option)
	Socket.Request("area.activityFavorHandler.dailyDrawInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastDailyDrawInfoResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.dailyDrawInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.dailyDrawInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.dailyDrawInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, dailyDrawInfoRequestEncoder, dailyDrawInfoRequestDecoder)
end


local function dailyDrawRequestEncoder(msg)
	local input = activityFavorHandler_pb.DailyDrawRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function dailyDrawRequestDecoder(stream)
	local res = activityFavorHandler_pb.DailyDrawResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.dailyDrawRequest(timeType,costType,cb,option)
	local msg = {}
	msg.timeType = timeType
	msg.costType = costType
	Socket.OnRequestStart("area.activityFavorHandler.dailyDrawRequest", option)
	Socket.Request("area.activityFavorHandler.dailyDrawRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastDailyDrawResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.dailyDrawRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.dailyDrawRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.dailyDrawRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, dailyDrawRequestEncoder, dailyDrawRequestDecoder)
end


local function recoveredInfoRequestEncoder(msg)
	local input = activityFavorHandler_pb.RecoveredInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function recoveredInfoRequestDecoder(stream)
	local res = activityFavorHandler_pb.RecoveredInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.recoveredInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.activityFavorHandler.recoveredInfoRequest", option)
	Socket.Request("area.activityFavorHandler.recoveredInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastRecoveredInfoResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.recoveredInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.recoveredInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.recoveredInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, recoveredInfoRequestEncoder, recoveredInfoRequestDecoder)
end


local function recoveredRequestEncoder(msg)
	local input = activityFavorHandler_pb.RecoveredRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function recoveredRequestDecoder(stream)
	local res = activityFavorHandler_pb.RecoveredResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.recoveredRequest(id,type,cb,option)
	local msg = {}
	msg.id = id
	msg.type = type
	Socket.OnRequestStart("area.activityFavorHandler.recoveredRequest", option)
	Socket.Request("area.activityFavorHandler.recoveredRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityFavorHandler.lastRecoveredResponse = res
			Socket.OnRequestEnd("area.activityFavorHandler.recoveredRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.activityFavorHandler.recoveredRequest decode error!!"
			end
			Socket.OnRequestEnd("area.activityFavorHandler.recoveredRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, recoveredRequestEncoder, recoveredRequestDecoder)
end


local function superPackageBuyPushDecoder(stream)
	local res = activityFavorHandler_pb.SuperPackageBuyPush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityFavorHandler.superPackageBuyPush(cb)
	Socket.On("area.activityFavorPush.superPackageBuyPush", function(res) 
		Pomelo.ActivityFavorHandler.lastSuperPackageBuyPush = res
		cb(nil,res) 
	end, superPackageBuyPushDecoder) 
end




--------------------------------Pomelo-------------------------------
return Pomelo
